pypdf/filters.py

```diff
@@ -847,7 +847,7 @@ def bits2byte(data: bytes, size: Tuple[int, int], bits: int) -> bytes:
         lookup: Any
         base: Any
         hival: Any
-        if isinstance(color_space, ArrayObject) and color_space[0] == "/Indexed":
+        if isinstance(color_space, ArrayObject) and len(color_space) == 4 and color_space[0] == "/Indexed":
```

Comment:
How is this change related to the CLI?

Reply:
This is a fixup for a bug.

Introducing this CLI and the related unit tests revealed this issue:
```
  File "/opt/pypdf/pypdf/filters.py", line 851, in _handle_flate
    color_space, base, hival, lookup = (
ValueError: not enough values to unpack (expected 4, got 3)
```

There is also another bug spotted there:
https://github.com/py-pdf/pypdf/blob/issue-2160/tests/test_cli.py#L91

Reply:
@Lucas-C 
can you provide the pdf showing the issue

Reply:
Yes, this is one of `py-pdf` test samples: https://github.com/py-pdf/pypdf/blob/main/resources/AutoCad_Diagram.pdf

The error was raised by simply calling `pypdf extract-images AutoCad_Diagram.pdf`

Reply:
I reported the 2nd bug there: https://github.com/py-pdf/pypdf/issues/2176

Should I open a distinct issue for the 1st bug, regarding `filters.py` & `color_space`?